plugin/scratch_plugin.lua

```diff
@@ -4,25 +4,53 @@ if vim.g.loaded_scratch == 1 then
   return
 end
 vim.g.loaded_scratch = 1
+vim.g.os_sep = vim.g.os_sep or vim.fn.has("win32") and "\\" or "/"
 
 -- create any global command that does not depend on user setup
 -- usually it is better to define most commands/mappings in the setup function
 -- Be careful to not overuse this file!
 
 -- TODO: remove those requires
-local scratch_api = require("scratch.api")
 
-local scratch_main = require("scratch")
-scratch_main.setup()
+local base_path = vim.fn.stdpath("cache") .. vim.g.os_sep .. "scratch.nvim" .. vim.g.os_sep
+---@type Scratch.Actor
+local default_config = {
+  scratch_file_dir = base_path, -- where your scratch files will be put
+  filetypes = { "lua", "js", "py", "sh" }, -- you can simply put filetype here
+  win_config = {
+    relative = "editor", -- Assuming you want the floating window relative to the editor
+    row = 2,
+    col = 5,
+    width = vim.api.nvim_win_get_width(0) - 10, -- Get the screen width - row * col
+    --api_get_option("lines") - 5,
+    height = vim.api.nvim_win_get_height(0) - 5, -- Get the screen height - col
+    style = "minimal",
+    border = "single",
+    title = "",
+  },
+  file_picker = "fzflua",
+  filetype_details = {},
+  localKeys = {},
+  manual_text = "MANUAL_INPUT",
+}
+local Actor = require("scratch.actor")
+---@type Scratch.Actor
+vim.g.scratch_actor = vim.g.scratch_actor or setmetatable(default_config, Actor)
 
 vim.api.nvim_create_user_command("Scratch", function(args)
   if args.range > 0 then
-    scratch_api.scratch({ content = utils.getSelectedText() })
+    vim.g.scratch_actor:scratch({ content = utils.getSelectedText() })
   else
-    scratch_api.scratch()
+    vim.g.scratch_actor:scratch({})
   end
 end, { range = true })
 
-vim.api.nvim_create_user_command("ScratchOpen", scratch_api.openScratch, {})
-vim.api.nvim_create_user_command("ScratchOpenFzf", scratch_api.fzfScratch, {})
-vim.api.nvim_create_user_command("ScratchWithName", scratch_api.scratchWithName, {})
+vim.api.nvim_create_user_command("ScratchOpen", function()
+  require("scratch.default_finder").findByNative(vim.g.scratch_actor.scratch_file_dir)
+end, {})
+vim.api.nvim_create_user_command("ScratchOpenFzf", function()
```

Comment:
`ScratchOpenFzf` should `Grep the contents of scratch files and open a selected file.`

Maybe I should rename it to `ScratchOpenGrep`